src/schemathesis/_hypothesis.py

```diff
@@ -24,11 +24,17 @@
 
 
 def create_test(
-    endpoint: Endpoint, test: Callable, settings: Optional[hypothesis.settings] = None, seed: Optional[int] = None
+    endpoint: Endpoint,
+    test: Callable,
+    settings: Optional[hypothesis.settings] = None,
+    seed: Optional[int] = None,
+    before_one_hook: Optional[Callable] = None,
```

Comment:
I am thinking if we can make `create_test` a generator, so it always generates tests, in this way we won't have to iterate one extra time on the callee side as it is now in `_get_tests_with_hooks` and `get_all_tests`. Also, we will not need to pass `before_one_hook` argument here

Maybe, in this case, we can also rename `create_test` to something more appropriate

Reply:
I made `create_test` a generator. 

However, because `create_test` may throw an `InvalidSchema` exception, all the generated values must be consumed and put in a list to see if an `InvalidSchema` exception is thrown. I also wrap the `InvalidSchema` exception in a list, so we may still iterate on the return value from `make_test_or_exception`.

I will push the changes, so you can see the code (most likely today). I'm wondering if it makes more sense for `create_test` to return a list instead of a generator.

Reply:
I think that this generator will be consumed anyway in a few next steps, since `collect` inside the pytest plugin will consume it before proceeding, so it will be done during the collection phase anyway. Currently, if there are e.g. 5 generated tests and an error happens, then all 5 will be replaced with one lambda. I am not sure, but there could be 4 valid tests that can be executed - maybe it would be better to check each instance during the iteration, and if it is failing then generate that lambda with `pytest.fail`. What do you think?